spot_driver/spot_driver/spot_ros2.py

```diff
@@ -2913,8 +2913,11 @@ def step(self) -> None:
     def destroy_node(self) -> None:
         self.get_logger().info("Shutting down ROS driver for Spot")
         if self.spot_wrapper is not None:
-            if self.spot_wrapper.check_is_powered_on():
+            # If the driver owns the estop, we want to sit before powering off.
+            if self.spot_wrapper.check_is_powered_on() and self.start_estop.value:
                 self.spot_wrapper.sit()
+                # This sleep is necessary for the robot to not collapse
+                time.sleep(3.0)
```

Comment:
Unsure what to do to better address this besides this hardcoded `sleep`. The `sit()` command returns before the robot finishes sitting, even though this is a synchronous call under the hood

Reply:
The spot-sdk doesn't really have synchronous calls. They have some blocking function that basically poll on requested feedback until a certain state. Should probably do that here instead of the blind sleep.

Also in the case we do not have the estop, we should probably still send the `stop` command, otherwise if spot is executing a trajectory it will continue to do so.

Reply:
Will it?  Cuz it'll lose the lease.  We should test that.

Reply:
@jbarry-bdai will test that out this morning

Reply:
@amessing-bdai is there an example of how to do that somewhere else in this repo (or spot wrapper)?

Reply:
> Will it? Cuz it'll lose the lease. We should test that.

I don't think it matters that you have lost the lease if you have already sent the trajectory. We have had this happen during extract in the wud demo.

Reply:
@khughes-bdai https://github.com/boston-dynamics/spot-sdk/blob/8f781a87b8816168ec59415db853934f41568620/python/bosdyn-client/src/bosdyn/client/robot_command.py#L2064

or any of the `block_until` functions in the spot-sdk

Reply:
The robot _sometimes_ sits down with the addition of the `stop` command. Additionally I am noticing lease errors if I kill the driver in the middle of an action, then start it again with this change (not consistently, but often) which makes me hesitant to add this in:
```
[spot_ros2-1]     raise Exception(err_msg)
[spot_ros2-1] Exception: Lease on "body" has state (Status.OTHER_OWNER)
[spot_ros2-1] [WARN] [1719413452.923502086] [Rex.spot_ros2]: Goal state not set, assuming aborted. Goal ID: [ 66 193 196 138 203  85  71 105 185 101 203 122  49  76 163  75]

```

Reply:
Can you return the lease as we close?

Reply:
Isn't that accounted for in `spot_wrapper.isconnect()`? This internally calls `spot_wrapper.releaseLease()`. Or is returning the lease different from releasing it?

Reply:
Ah, that returns the lease internally. I am surprised you are getting the "other owner" error. Might be a BD support time

Reply:
Put up https://github.com/bdaiinstitute/spot_wrapper/pull/122 and this will make it so that we no longer need to have the `sleep`. I can make a ticket for looking into the `stop` command and the corresponding lease issues some more, might need to address this separately